mcp_zammad/server.py

```
@@ -221,14 +242,32 @@ def _truncate_json_response(content: str, obj: dict[str, Any], limit: int) -> st
+    if largest_key:
+        while obj[largest_key] and len(json_str) > limit:  # type: ignore[index]
+            obj[largest_key].pop()  # type: ignore[index]
             json_str = _serialize_json(obj, use_compact=use_compact)
 
-    return _serialize_json(obj, use_compact=use_compact)
+    # Hard-cap fallback: if still over limit, convert to plaintext to avoid invalid JSON
+    if len(json_str) > limit:
+        # Convert to plaintext summary to avoid producing invalid JSON fragments
+        try:
+            plaintext_repr = json.dumps(obj, ensure_ascii=False)
+        except Exception:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)